connectors/source.py

```diff
@@ -103,6 +105,9 @@ def get_fields(self):
     def is_empty(self):
         return len(self._config) == 0
 
+    def to_dict(self):
+        return deepcopy(self._raw_config)
```

Comment:
`deepcopy` is overkill, even `copy` as long as the config is mappings of mappings -- you can use `dict`

```
>>> o = {'a': {'b': 'c'}}
>>> dict(o)
{'a': {'b': 'c'}}
```